public/app/core/services/backend_srv.ts

```diff
@@ -457,121 +152,211 @@ export class BackendSrv implements BackendService {
-    resultType: CancellationType
-  ): MonoTypeOperatorFunction<FetchResponse | DataSourceSuccessResponse | SuccessResponse> => inputStream =>
-    inputStream.pipe(
-      takeUntil(
-        this.inFlightRequests.pipe(
-          filter(requestId => {
-            let cancelRequest = false;
-
-            if (options && options.requestId && options.requestId === requestId) {
-              // when a new requestId is started it will be published to inFlightRequests
-              // if a previous long running request that hasn't finished yet has the same requestId
-              // we need to cancel that request
-              cancelRequest = true;
-            }
-
-            if (requestId === CANCEL_ALL_REQUESTS_REQUEST_ID) {
-              cancelRequest = true;
-            }
-
-            return cancelRequest;
-          })
+              return throwError(error);
+            })
+          )
         )
-      ),
-      // when a request is cancelled by takeUntil it will complete without emitting anything so we use throwIfEmpty to identify this case
-      // in throwIfEmpty we'll then throw an cancelled error and then we'll return the correct result in the catchError or rethrow
-      throwIfEmpty(() => ({
-        cancelled: true,
-      })),
-      catchError(err => {
-        if (!err.cancelled) {
-          return throwError(err);
+      );
+  }
+
+  showApplicationErrorAlert(err: FetchError) {
+    let description = '';
+    let message = err.data.message;
+
+    if (message.length > 80) {
+      description = message;
+      message = 'Error';
+    }
+
+    // Validation
+    if (err.status === 422) {
+      message = 'Validation failed';
+    }
+
+    this.dependencies.appEvents.emit(err.status < 500 ? AppEvents.alertWarning : AppEvents.alertError, [
+      message,
+      description,
+    ]);
+  }
+
+  showSuccessAlert<T>(response: FetchResponse<T>) {
+    const { config } = response;
+
+    if (config.method === 'GET' || config.showSuccessAlert === false) {
+      return;
+    }
+
+    // Skip success alerts for data queries
+    if (isDataQuery(response.config.url)) {
+      return;
+    }
+
+    const data: { message: string } = response.data as any;
+
+    if (data?.message) {
+      this.dependencies.appEvents.emit(AppEvents.alertSuccess, [data.message]);
+    }
+  }
+
+  processRequestError(options: BackendSrvRequest, err: FetchError): FetchError {
+    err.data = err.data ?? { message: 'Unexpected error' };
+
+    if (typeof err.data === 'string') {
+      err.data = {
+        error: err.statusText,
+        response: err.data,
+        message: err.data,
+      };
+    }
+
+    // If no message but got error string, copy to message prop
+    if (err.data && !err.data.message && typeof err.data.error === 'string') {
+      err.data.message = err.data.error;
+    }
+
+    // check if we should show an error alert
+    if (err.data.message && !isDataQuery(options.url) && options.showErrorAlert !== false) {
```

Comment:
no that is not the same, since if it's undefined we should show error alert. pretty poorly constructed boolean, should be named noErrorAlert and be set to true to disable default behavior but we had a showSuccessAlert already that was done like this. Will look at inverting that but could break some plugins that rely on it

Reply:
I see. Well, if you decide to stick with as it is, then I think it would be a good idea to be more explicit about the behavior, so that people not familiar with it can't make the same mistake as I did. ;)
=> `options.showErrorAlert === undefined || options.showErrorAlert`

Reply:
Improved the logic for showErrorAlert and showSuccessAlert, now in the code false mean one thing and undefined means "auto mode"  (and true means force to true). Before the logic was only if false disable auto logic, so there was no way to enable it for cases where auto logic made it not show.